apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx

```diff
@@ -69,6 +71,13 @@ const UsernameTextfield = (props: ICustomUsernameProps & Partial<React.Component
       onSuccessMutation && (await onSuccessMutation());
       setOpenDialogSaveUsername(false);
       setCurrentUsername(inputUsernameValue);
+      await update({
+        ...session,
+        user: {
+          ...session?.user,
+          username: inputUsernameValue,
+        },
+      });
```

Comment:
We can also just use this and let the server revalidate, otherwise we would need to validate the input again.

```suggestion
      await update();
```